src/book/02-system/02-subsystems/04-motion.mdx

```diff
@@ -5,3 +5,66 @@ title: Motion
 description: Motion and how it works in the NUbots codebase.
 slug: /system/subsystems/motion
 ---
+
+# Walk Engine
+
+Currently NUbots has two walk engines. One is an **open loop hybrid walk controller** and the other is an **open loop walk** based on **quintic splines**. These engines take care of the mathematics and control involved in taking steps without falling over. The walk engines do not plan where to walk to, only how to walk.
+
+A walk engine will trigger when a [**walkcommand**](https://github.com/NUbots/NUbots/blob/master/shared/message/motion/WalkCommand.proto) is [emitted](/system/nubots/nuclear#emit-statements). The walk command gives a vector [x, y, z] which represents:
+
+| Variable | Description                   |
+| -------- | ----------------------------- |
+| x        | velocity in metres per second |
+| y        | velocity in metres per second |
+| z        | angle in radians per second   |
```

Comment:
Good point. I guess it would be described as yaw with 0 being in the direction of the x-axis of the robot. Positive would go counter-clockwise. I'll add that in.